p/scripts/main.js

```diff
@@ -1215,6 +1215,84 @@ function init_notifications() {
 }
 // </notification>
 
+// <popup>
+var popup = null,
+	popup_iframe = null,
+	popup_p = null,
+	popup_working = false;
+
+function openPopupWithMessage(msg) {
+	if (popup_working === true) {
+		return false;
+	}
+
+	popup_working = true;
+
+	popup_p.innerHTML = msg;
+	popup_p.removeAttribute('hidden');
+
+	popup.style.display = "block";
+}
+
+function openPopupWithSource(source) {
+	if (popup_working === true) {
+		return false;
+	}
+
+	popup_working = true;
+
+	popup_iframe.src = source;
+	popup_iframe.removeAttribute('hidden');
+
+	popup.style.display = "block";
+}
+
+function closePopup() {
+	popup.style.display = "none";
+
+	popup_iframe.src = 'about:blank';
+	popup_iframe.setAttribute('hidden', true);
+
+	popup_p.setAttribute('hidden', true);
+
+	popup_working = false;
+}
+
+function init_popup() {
+	// Fetch elements.
+	popup = document.getElementById('popup');
+	popup_iframe = document.getElementById('popup-iframe');
+	popup_p = document.getElementById('popup-p');
+
+	//Configure clode-button.
+	document.getElementsByClassName("popup-close")[0].onclick = function() {
```

Comment:
```suggestion
	document.getElementsByClassName('popup-close')[0].onclick = function() {
```

Also this kind of thing may be a bit undependable, although nowadays the DOM between browsers is probably more stable than it used to be. It's better practice to get the relevant child node more dependably than just a random mystery index.

Reply:
Not sure how to do that :s

Reply:
It's more of a general comment really. Since it's 2020 you can get away with just doing `document.querySelector('.popup-close').addEventListener('click', …` which is really just the same thing as the 0 index anyway.

Reply:
Okay, I will use that.